src/rustdoc-json-types/lib.rs

```diff
@@ -329,7 +329,7 @@ pub struct GenericParamDef {
 #[serde(rename_all = "snake_case")]
 pub enum GenericParamDefKind {
     Lifetime { outlives: Vec<String> },
-    Type { bounds: Vec<GenericBound>, default: Option<Type> },
+    Type { bounds: Vec<GenericBound>, default: Option<Type>, synthetic: bool },
```

Comment:
This should be documented for what it means, because I can't find any reference to synthetic generic params outside the compiller.

Reply:
Good point. I added docs now. I considered documenting other fields, but I think the structure and naming is pretty clear by themselves to someone that is familiar with the Rust type system. The diff is a bit big due to rustfmt though.

Also, big thanks for making the test much more robust.

It is also worth thinking about if we instead of adding `synthetic` should simply omit synthetic params from JSON. But, I suspect that if we go that route, that will cascade into current or future problems, such as the need to special case other code that relies on synthetic generic args being available.

Analogous to how it appears to be the case that omitting some private modules was a mistake (see #93518)

Reply:
> It is also worth thinking about if we instead of adding synthetic should simply omit synthetic params from JSON.

Yeah, the long term plan is to make params only be the names, and put all constraints in the `where_predicates` (https://github.com/rust-lang/rust/pull/93803). But thats a larger change and we can land this now, to also improve the shape of the tests for when it lands